### PR TITLE
Revert "Open links externally instead of in the html editor in etsdemo application"

### PR DESCRIPTION
**Pull request by kitchoi**
Reverts enthought/traitsui#1446

Unfortunately this does not work well with PySide2 (with QtWebEngine) due to #1448.

### CODE REVIEW

**Review by rahulporuri — APPROVED** [review]
LGTM

**Comment by corranwebster** [comment]
This is the wrong fix - open links externally is the correct behaviour, since we don't have the full browser navigation available in a simple html view.  The bug is not with the demo app, but with the way that we are hacking in WebEngine support in Pyface; and this issue will affect every web view which uses this option.

Short-term fix is to have the HTML editor detect which API call is available and use the right one.

The right fix is to spend the effort to get Pyface's web support into a sane state (it's becoming increasingly clear that WebEngine _isn't_ a drop-in replacement for WebKit) and then have the Qt HTMLEditor do some sort of mapping of APIs (or present a lower-level common API in Pyface or something).

**Comment by kitchoi** [comment]
> This is the wrong fix - open links externally is the correct behaviour, since we don't have the full browser navigation available in a simple html view. The bug is not with the demo app, but with the way that we are hacking in WebEngine support in Pyface; and this issue will affect every web view which uses this option.
>
> Short-term fix is to have the HTML editor detect which API call is available and use the right one.
>
> The right fix is to spend the effort to get Pyface's web support into a sane state (it's becoming increasingly clear that WebEngine isn't a drop-in replacement for WebKit) and then have the Qt HTMLEditor do some sort of mapping of APIs (or present a lower-level common API in Pyface or something).

Agreed that the HTML editor needs to be fixed. This PR does not attempt to close or fix #1448.
However IMO, the behaviour about opening a link is really really minor in comparison with the lost of the entire view and crashing of the etsdemo application. We can resurrect this flag once HTML editor is fixed and released.

I will leave this to @rahulporuri and @corranwebster whether to merge.

**Comment by kitchoi** [comment]
For reference, #1450 is blocked by this PR, and any attempt to add tests for meaningfully test the etsdemo view will be blocked by this error as well.

**Comment by kitchoi** [comment]
> I will leave this to @rahulporuri and @corranwebster whether to merge.

Actually, it might be best to flip the option :D 
If this should not be merged, please go ahead close this PR. Otherwise I will merge this later in my next free cycle to avoid PRs going stale and unresolvable.

**Comment by corranwebster** [comment]
I have a draft PR that should do the quick hacky fix (but is completely untested, and I don't have time to finish it properly).

**Comment by corranwebster** [comment]
Closing in favour of #1451 or something along those lines.

**Pull request closed without merging**
